### PR TITLE
feat(coinjoin): pick privacy level

### PR DESCRIPTION
**Pull request by Syn-McJ**
The user must be able to pick privacy level for CoinJoin mixing.

## Issue being fixed or feature implemented
- Repurpose the `CreateUsernamePrivacyFragment` for the new flow.
- Adjust the UI and logic to match the latest design.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/service/CoinJoinService.kt [hunk 1]
```diff
@@ -66,9 +66,10 @@ enum class CoinJoinMode {
 }
 
 interface CoinJoinService {
+    var mode: CoinJoinMode
+    val mixingStatus: MixingStatus
+
     fun needsToMix(amount: Coin): Boolean
-    fun getMode(): CoinJoinMode
-    fun setMode(mode: CoinJoinMode)
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/service/CoinJoinService.kt:71]: I think in Kotlin we don't need dedicated setter/getter methods

File: wallet/src/de/schildbach/wallet/ui/coinjoin/CoinJoinLevelViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            AnalyticsConstants.CoinJoinPrivacy.USERNAME_PRIVACY_BTN_CONTINUE,
+            mapOf(AnalyticsConstants.Parameter.VALUE to mode.name)
+        )
+
+        coinJoinService.mode = mode
+        coinJoinService.prepareAndStartMixing() // TODO: change the logic if needed
+    }
+
+    suspend fun stopMixing() {
+//        coinJoinService.stopMixing() // TODO expose stop method?
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/coinjoin/CoinJoinLevelViewModel.kt:63]: Stopping the mixing isn't exposed in the `coinJoinService`, don't know if that's what we want. I'll leave this to you @HashEngineering

File: wallet/src/de/schildbach/wallet/service/CoinJoinService.kt (hunk 1 above)
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/service/CoinJoinService.kt:71]: Yeah, this was probably due to thinking in Java.

File: wallet/src/de/schildbach/wallet/ui/coinjoin/CoinJoinLevelViewModel.kt (hunk 2 above)
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/coinjoin/CoinJoinLevelViewModel.kt:63]: Yes, we will have to change how we tell **CoinJoinService** what to do.  With this PR we are adding a means for the user to stop the process.  There will be other events that will activate or deactivate the CoinJoinService was well.

**Review by HashEngineering — APPROVED** [review]

**Pull request merged**